tests/std/tests/P0288R9_move_only_function/test.cpp

```diff
@@ -0,0 +1,393 @@
+            return z - 3;
+        }
+    };
+
+    move_only_function<int(s_t*, int)> mem_fun_ptr(&s_t::f);
+    move_only_function<int(s_t*)> mem_ptr(&s_t::j);
+    move_only_function<int(int)> fun_ptr(&s_t::g);
+
+    s_t s;
+    assert(mem_fun_ptr);
+    assert(mem_fun_ptr(&s, 3) == 5);
+    assert(mem_ptr);
+    assert(mem_ptr(&s) == 6);
+    assert(fun_ptr);
+    assert(fun_ptr(34) == 31);
+
+    move_only_function<int(s_t*, int)> mem_fun_ptr_n(static_cast<decltype(&s_t::f)>(nullptr));
+    move_only_function<int(s_t*)> mem_ptr_n(static_cast<decltype(&s_t::j)>(nullptr));
+    move_only_function<int(int)> fun_ptr_n(static_cast<decltype(&s_t::g)>(nullptr));
+
+    assert(!mem_fun_ptr_n);
+    assert(!mem_ptr_n);
+    assert(!fun_ptr_n);
+}
+
+template <bool Nx>
+struct test_noexcept_t {
+    int operator()() noexcept(Nx) {
+        return 888;
+    }
+};
+
+void test_noexcept() {
+    using f_x  = move_only_function<int()>;
+    using f_nx = move_only_function<int() noexcept>;
+
+    static_assert(!noexcept(declval<f_x>()()));
+#ifdef __cpp_noexcept_function_type
+    static_assert(noexcept(declval<f_nx>()()));
+#else // ^^^ defined(__cpp_noexcept_function_type) ^^^ / vvv !defined(__cpp_noexcept_function_type) vvv
+    static_assert(!noexcept(declval<f_nx>()()));
+#endif // ^^^ !defined(__cpp_noexcept_function_type) ^^^
+
+    static_assert(is_constructible_v<f_x, test_noexcept_t<false>>);
+    assert(f_x(test_noexcept_t<false>{})() == 888);
+
+    static_assert(is_constructible_v<f_x, test_noexcept_t<true>>);
+    assert(f_x(test_noexcept_t<true>{})() == 888);
+
+#ifdef __cpp_noexcept_function_type
+    static_assert(!is_constructible_v<f_nx, test_noexcept_t<false>>);
+#else // ^^^ defined(__cpp_noexcept_function_type) ^^^ / vvv !defined(__cpp_noexcept_function_type) vvv
+    static_assert(is_constructible_v<f_nx, test_noexcept_t<false>>);
+    assert(f_nx(test_noexcept_t<false>{})() == 888);
+#endif // ^^^ !defined(__cpp_noexcept_function_type) ^^^
+
+    static_assert(is_constructible_v<f_nx, test_noexcept_t<true>>);
+    assert(f_nx(test_noexcept_t<true>{})() == 888);
+}
+
+template <bool>
+struct test_const_t {
+    int operator()() {
+        return 456;
+    }
+};
+
+template <>
+struct test_const_t<true> {
+    int operator()() const {
+        return 456;
+    }
+};
+
+void test_const() {
+    using f_c  = move_only_function<int() const>;
+    using f_nc = move_only_function<int()>;
+
+    static_assert(is_constructible_v<f_nc, test_const_t<false>>);
+    f_nc f1(test_const_t<false>{});
+    assert(f1() == 456);
+
+    static_assert(is_constructible_v<f_nc, test_const_t<true>>);
+    f_nc f2(test_const_t<true>{});
+    assert(f2() == 456);
+
+    static_assert(!is_constructible_v<f_c, test_const_t<false>>);
+
+    static_assert(is_constructible_v<f_c, test_const_t<true>>);
+    f_c f3(test_const_t<true>{});
+    assert(f3() == 456);
+    const f_c f4(test_const_t<true>{});
+    assert(f4() == 456);
+}
+
+int main() {

```

Comment:
There appears to be no test coverage of construction from `nullptr` (only of construction from null function/member pointers).

Similarly, there's no coverage of assignment from `nullptr`.